types/sass/index.d.ts

```diff
@@ -153,12 +153,18 @@ export interface Options {
     logger?: Logger;
 }
 
+export interface LoggerOptions {
+    deprecation: boolean;
+    span?: SourceSpan;
+    stack?: string;
+}
+
 export interface Logger {
     /** This method is called when Sass emits a debug message due to a @debug rule. */
-    debug?(message: string, options: { deprecation: boolean; span?: SourceSpan; stack?: string }): void;
+    debug?(message: string, options: LoggerOptions): void;
 
     /** This method is called when Sass emits a debug message due to a @warn rule. */
-    warn?(message: string, options: { span: SourceSpan }): void;
+    warn?(message: string, options: LoggerOptions): void;
```

Comment:
Note sure if it warrants a seperate interface. 

 ```suggestion
  warn?(
    message: string,
    options: {
      deprecation: boolean;
      span?: SourceSpan;
      stack?: string;
    }
  ): void;
```